posts/2021-08-13-dev-mode-multi-module.adoc

```diff
@@ -0,0 +1,130 @@
+
+The following example is the link:https://github.com/OpenLiberty/guide-maven-multimodules[Open Liberty multi-module sample application]. This project contains three modules: a Java library in the `jar` folder, a web module in the `war` folder, and template files and the Liberty server configuration in the `ear` folder. 
+
+----
+guide-maven-multimodules
+|_ pom.xml
+|_ jar
+    |_pom.xml
+    |_src/main/java
+|_ war
+    |_pom.xml
+    |_src/main
+        |_java
+        |_webapp
+|_ ear
+    |_pom.xml
+    |_src
+        |_main/liberty/config
+        |_test/java
+----
+
+Each module is defined in the top-level aggregator `pom.xml` file:
+
+[source,xml]
+----
+    <modules>
+        <module>jar</module>
+        <module>war</module>
+        <module>ear</module>
+    </modules>
+----
+
+The modules within the project can also have dependencies on each other. In this example, the `war` module depends on the `jar` module, and the `ear` module depends on both the `war` and `jar` modules. By leveraging dev mode for multiple modules, not only will the code changes you make be automatically picked up and hot deployed, but any dependent modules will also be recompiled and hot deployed.
+
+== Running Liberty on a multi-module project
+
+Liberty can be run on a multi-module project with a single command by using dev mode, dev mode with container support, or by starting the server in the foreground.  To do any of these, the Liberty Maven plug-in must first be defined.
+
+=== Defining the Liberty Maven plug-in
+
+To use the Liberty Maven plug-in in your multi-module project, you can define the plug-in in either the parent `pom.xml` file of every module or in the `pom.xml` of every module.
+
+For example, if your modules have a `<parent>` element that points to a parent project, add the Liberty Maven plug-in to the `<pluginManagement>` section of the parent project's `pom.xml` as follows:
+[source,xml]
+----
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.openliberty.tools</groupId>
+                    <artifactId>liberty-maven-plugin</artifactId>
+                    <version>3.4</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+----
+
+=== Starting dev mode
+
+To start dev mode, from the directory containing the top-level multi-module `pom.xml`, run:
+
+----
+mvn liberty:dev
+----
+
+Once dev mode starts successfully, changes in all modules are detected and hot deployed. By default, any dependent modules will also be recompiled and deployed. For example, if a change is made in the `jar` module of the sample application described above, the `war` module will also be recompiled since it depends on the `jar` module. 
```

Comment:
By default, any dependent modules will also be recompiled and deployed. For example, if a change is made in the `jar` module of the sample application described above, the `war` module will also be recompiled since it depends on the `jar` module. 

---->

By default, any dependent modules are also recompiled and deployed. For example, if you make a change in the `jar` module of the previously described sample application, the `war` module is also recompiled because it depends on the `jar` module.